dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/datapoints/AssembledDataMigrationTrigger.kt

```diff
@@ -0,0 +1,87 @@
+package org.dataland.datalandbackend.services.datapoints
+
+import org.dataland.datalandapikeymanager.openApiClient.model.ApiKeyMetaInfo
+import org.dataland.datalandbackend.repositories.DataMetaInformationRepository
+import org.dataland.datalandinternalstorage.openApiClient.api.ActuatorApi
+import org.dataland.keycloakAdapter.auth.DatalandApiKeyAuthentication
+import org.dataland.specificationservice.openApiClient.api.SpecificationControllerApi
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.event.ContextRefreshedEvent
+import org.springframework.context.event.EventListener
+import org.springframework.scheduling.annotation.Async
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+/**
+ * Triggers the migration of stored datasets to assembled datasets when the backend starts
+ */
+@Service
+class AssembledDataMigrationTrigger
+    @Autowired
+    constructor(
+        private val assembledDataMigrationManager: AssembledDataMigrationManager,
+        private val actuatorApi: ActuatorApi,
+        private val specificationControllerApi: SpecificationControllerApi,
+        private val dataMetaInformationRepository: DataMetaInformationRepository,
+    ) {
+        companion object {
+            const val TRIES_TO_WAIT_FOR_INTERNAL_STORAGE = 10
+            const val WAIT_TIME_FOR_INTERNAL_STORAGE = 10_000L
+        }
+
+        private val logger = LoggerFactory.getLogger(javaClass)
+
+        private fun waitUntilInternalStorageHealthy() {
+            Thread.sleep(WAIT_TIME_FOR_INTERNAL_STORAGE)
+            for (attempt in 1..TRIES_TO_WAIT_FOR_INTERNAL_STORAGE) {
+                @Suppress("TooGenericExceptionCaught")
+                try {
+                    actuatorApi.health()
+                    return
+                } catch (e: Exception) {
```

Comment:
Wenn ich mich richtig erinnere kann man exceptions auch ignorieren in detekt indem man einfach einen speziellen Namen vergibt sowas wie "ignore" oder so. Dann müsste man das suppress weglassen können und müsste die exception nicht verwenden. So müsste das aussehen "} catch (ignore: Exception) {"

Reply:
Fixed